### DISCUSSION TITLE
Assistance with cloudflare.snip

### DISCUSSION THREAD

**Discussion by pallebone** [ubergeek77/Lemmy-Easy-Deploy#58] [General]
Hi there,

The cloudflare.snip file is in this format:

(cloudflare_https) {
  tls {
    dns cloudflare {env.CF_API_TOKEN}
  }
}

For CF_API_TOKEN how can I create this exactly?
Can I create an API key or am I having to use the global api key? I would prefer to create a key if this is possible.

Hope you know how to proceed.

Once creating and putting the file into the custom directory I can redeploy to test.

Kind regards
Peter

**Comment by ubergeek77** [comment]
You just need to make a token with the "Edit Zone DNS" template like this:

https://developers.cloudflare.com/fundamentals/api/get-started/create-token/

No need to make a copy of .snip since you aren't changing it, it seems.

The key goes in config.env.

**Comment by pallebone** [comment]
"The key goes in config.env."

Oh ok thank you. This makes sense. So no need for anything other than editing config.env then.

Will try now.

**Comment by pallebone** [comment]
When I redeployed it was it supposed to fetch a new caddy image? Because it did not do this and the site isnt working 100%. I will continue to try troubleshoot.

**Comment by ubergeek77** [comment]
It's supposed to build a new caddy image locally and replace the caddy image in docker-compose.yml accordingly

**Comment by pallebone** [comment]
Can I force that somehow?

**Comment by pallebone** [comment]
The image in the docker compose stayed as:
image: caddy:latest

**Comment by ubergeek77** [comment]
You can't force it, it will happen as needed if the Cloudflare API token is specified, and only on redeploys. If you didn't redeploy with `-f` then essentially the script doesn't do anything

**Comment by pallebone** [comment]
Oh oops I didnt realise I had to use -f I will try again thank you.

**Comment by pallebone** [comment]
Thank you you are 100% this fixed it - you are an eternal legend. Thanks!!
